engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -194,13 +153,42 @@ public ValidatorFactoryImpl(ConfigurationState configurationState) {
 				getConstraintValidatorPayload( hibernateSpecificConfig )
 		);
 
-		this.constraintValidatorManager = new ConstraintValidatorManagerImpl(
+		ConstraintValidatorManager constraintValidatorManager = new ConstraintValidatorManagerImpl(
 				configurationState.getConstraintValidatorFactory(),
 				this.validatorFactoryScopedContext.getConstraintValidatorInitializationContext()
 		);
 
 		this.validationOrderGenerator = new ValidationOrderGenerator();
 
+		ValueExtractorManager valueExtractorManager = new ValueExtractorManager( configurationState.getValueExtractors() );
+		ConstraintHelper constraintHelper = new ConstraintHelper();
+		TypeResolutionHelper typeResolutionHelper = new TypeResolutionHelper();
+		JavaBeanHelper javaBeanHelper = new JavaBeanHelper( ValidatorFactoryConfigurationHelper.getGetterPropertySelectionStrategy( hibernateSpecificConfig, properties, externalClassLoader ) );
+
+		this.constraintCreationContext = new ConstraintCreationContext( constraintHelper, constraintValidatorManager, typeResolutionHelper, valueExtractorManager );
+
+		this.executableHelper = new ExecutableHelper( typeResolutionHelper );
+		this.javaBeanHelper = new JavaBeanHelper( ValidatorFactoryConfigurationHelper.getGetterPropertySelectionStrategy( hibernateSpecificConfig, properties, externalClassLoader ) );
```

Comment:
There's one `JavaBeanHelper` already created at Line 166. WE probably can use that one.

Reply:
Good catch, fixed.